devlib/target.py

```diff
@@ -440,12 +440,12 @@ def write_value(self, path, value, verify=True):
             output = self.read_value(path)
             if not output == value:
                 message = 'Could not set the value of {} to "{}" (read "{}")'.format(path, value, output)
-                raise TargetError(message)
+                raise TargetNonTransError(message)
 
     def reset(self):
         try:
             self.execute('reboot', as_root=self.needs_su, timeout=2)
-        except (TargetError, TimeoutError, subprocess.CalledProcessError):
+        except (DevlibTransError, subprocess.CalledProcessError):
```

Comment:
Here I assumed that if executing "reboot" raises a non-transient error, it is because the command fails (permissions, corrupted binary, command not available etc) or similar issues that needs to be fixed. The caller of reset() expects the target to actually do something. The reason transient errors are ignored is that it is quite common that the session will just hang when the reboot happens, so that is a "usual expected" behavior that we unfortunately cannot distinguish from the ethernet cable being plugged out. In an reasonably scaled environment, you can't look closely at all the logs so you may end up running for month a test suite without realizing you are never rebooting the board, which might impact some tests results in an unexpected way that is hard to trace down.